### PR TITLE
Add fiscal code type

### PR DESCRIPTION
**Pull request by kin0992**
Add a type that describes the fiscal code

### CODE REVIEW

**Review by datalek — APPROVED** [review]

File: openapi/index.yaml [hunk 1]
```diff
@@ -25,6 +25,11 @@ components:
           example: []
           items:
             $ref: "#/components/schemas/Error"
+    FiscalCodeType:
+      description: The citizen's fiscal code.
+      type: string
+      pattern: "^[A-Z]{6}[0-9LMNPQRSTUV]{2}[ABCDEHLMPRST][0-9LMNPQRSTUV]{2}[A-Z][0-9LMNPQRSTUV]{3}[A-Z]$"
+      example: "SPNDNL80R13C555X"
```
Comment (francescopersico) [inline openapi/index.yaml:32]: Please remove the example that refer to Danilo :)

File: openapi/index.yaml [hunk 2]
```diff
@@ -25,6 +25,11 @@ components:
           example: []
           items:
             $ref: "#/components/schemas/Error"
+    FiscalCodeType:
```
Comment (francescopersico) [inline openapi/index.yaml:28]: Why not just FiscalCode without **Type**?

**Review by francescopersico — CHANGES_REQUESTED** [review]

File: openapi/index.yaml (hunk 2 above)
Comment (francescopersico) [reply openapi/index.yaml:28]: To be really compliant to interoperability standard  the italian "codice fiscale" must be named TaxCode as referred here

https://github.com/italia/form-pa/blob/bc4da698dfa61d80f7435c283f0cb0e78b8aa099/public/definitions.yaml#L358

**Review by francescopersico — CHANGES_REQUESTED** [review]

File: openapi/index.yaml (hunk 2 above)
Comment (francescopersico) [reply openapi/index.yaml:28]: Also remember that this file is specific for the CloudGaaP project and not for the products of the catalog (like for example the document sharing service).

File: openapi/index.yaml (hunk 1 above)
Comment (kin0992) [reply openapi/index.yaml]: Changed!

File: openapi/index.yaml (hunk 2 above)
Comment (kin0992) [reply openapi/index.yaml]: Ok, we are not going to reference the TaxCode from anything external to CloudGaaP (e.g. document-sharing-service).

File: openapi/index.yaml (hunk 1 above)
Comment (kin0992) [reply openapi/index.yaml]: For the record, that tax code is used a lot: https://github.com/search?q=org%3Apagopa+SPNDNL80R13C555X&type=code

Indeed we copied that from https://github.com/pagopa/io-functions-commons/blob/8cf65e0ea3e043313ba6c62f58a61d3aedd340df/openapi/definitions.yaml#L360

**Comment by kin0992** [comment]
We don't need this structure at the moment, so we are closing this PR.

**Pull request closed without merging**